controllers/ipoibnetwork_controller.go

```diff
@@ -0,0 +1,151 @@
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	netattdefv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+
+	mellanoxcomv1alpha1 "github.com/Mellanox/network-operator/api/v1alpha1"
+	"github.com/Mellanox/network-operator/pkg/config"
+	"github.com/Mellanox/network-operator/pkg/consts"
+	"github.com/Mellanox/network-operator/pkg/state"
+	"github.com/Mellanox/network-operator/pkg/utils"
+)
+
+// IpoibNetworkReconciler reconciles a IPoIBNetwork object
+type IpoibNetworkReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+
+	stateManager state.Manager
+}
+
+// +kubebuilder:rbac:groups=mellanox.com,resources=ipobnetworks,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=mellanox.com,resources=ipobnetworks/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=mellanox.com,resources=*,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=k8s.cni.cncf.io,resources=*,verbs=*
+
+//nolint:dupl
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+func (r *IpoibNetworkReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	reqLogger := r.Log.WithValues("ipobnetwork", req.NamespacedName)
+	reqLogger.Info("Reconciling IPoIBNetwork")
+
+	// Fetch the IPoIBNetwork instance
+	instance := &mellanoxcomv1alpha1.IPoIBNetwork{}
+	err := r.Get(context.TODO(), req.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	managerStatus, err := r.stateManager.SyncState(instance, nil)
+	r.updateCrStatus(instance, managerStatus, err)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if managerStatus.Status != state.SyncStateReady {
+		return reconcile.Result{
+			RequeueAfter: time.Duration(config.FromEnv().Controller.RequeueTimeSeconds) * time.Second,
+		}, nil
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *IpoibNetworkReconciler) updateCrStatus(cr *mellanoxcomv1alpha1.IPoIBNetwork, status state.Results,
+	syncError error) {
+	cr.Status.State = mellanoxcomv1alpha1.State(status.StatesStatus[0].Status)
+	if syncError != nil {
+		cr.Status.Reason = syncError.Error()
+	}
+
+	if cr.Status.State == state.SyncStateReady {
+		netAttachDef := &netattdefv1.NetworkAttachmentDefinition{}
+		err := r.Get(context.TODO(),
```

Comment:
Maybe pass context from `Reconcile`?

Reply:
Let's do it in a separate PR in all our controllers